docs/open-api/lineage.yaml

```diff
@@ -0,0 +1,281 @@
+# under the License.
+
+---
+
+openapi: 3.0.2
+info:
+  title: OpenLineage
+  version: 2-0-2
+  description: OpenLineage is an open source **lineage and metadata collection API** for the data ecosystem.
+  license:
+    name: Apache 2.0
+    url: http://www.apache.org/licenses/LICENSE-2.0.html
+paths:
+  /api/lineage:
+    post:
+      summary: Send an event related to the state of a run
+      description: Updates a run state for a job.
+      operationId: postRunEvent
+      tags:
+        - OpenLineage
+      requestBody:
+        content:
+          application/json:
+            schema:
+              oneOf:
+                - $ref: '#/components/schemas/RunEvent'
+                - $ref: '#/components/schemas/DatasetEvent'
+                - $ref: '#/components/schemas/JobEvent'
+      responses:
+        200:
+          description: OK
+components:
+  schemas:
+    BaseEvent:
+      type: object
+      properties:
+        eventTime:
+          description: the time the event occurred at
+          type: string
+          format: date-time
+        producer:
+          description: URI identifying the producer of this metadata. For example this could be a git url with a given tag or sha
+          type: string
+          format: uri
+          example: https://github.com/OpenLineage/OpenLineage/blob/v1-0-0/client
+        schemaURL:
+          description: The JSON Pointer (https://tools.ietf.org/html/rfc6901) URL to the corresponding version of the schema definition for this RunEvent
+          type: string
+          format: uri
+          example: https://openlineage.io/spec/0-0-1/OpenLineage.json
+      required:
+        - eventTime
+        - producer
+        - schemaURL
+    BaseFacet:
+      description: all fields of the base facet are prefixed with _ to avoid name conflicts in facets
+      type: object
+      properties:
+        _producer:
+          description: URI identifying the producer of this metadata. For example this could be a git url with a given tag or sha
+          type: string
+          format: uri
+          example: https://github.com/OpenLineage/OpenLineage/blob/v1-0-0/client
+        _schemaURL:
+          description: The JSON Pointer (https://tools.ietf.org/html/rfc6901) URL to the corresponding version of the schema definition for this facet
+          type: string
+          format: uri
+          example: https://openlineage.io/spec/1-0-2/OpenLineage.json#/$defs/BaseFacet
+      additionalProperties: true
+      required:
+        - _producer
+        - _schemaURL
+    RunFacet:
+      description: A Run Facet
+      type: object
+      allOf:
+        - $ref: '#/components/schemas/BaseFacet'
+    Run:
+      type: object
+      properties:
+        runId:
+          description: The globally unique ID of the run associated with the job.
+          type: string
+          format: uuid
+        facets:
+          description: The run facets.
+          type: object
+          anyOf:
```

Comment:
Why this `anyOf` ?

Reply:
Maybe for better extensions or some historical reason?